stdlib/public/runtime/SymbolInfo.h

```diff
@@ -43,14 +43,74 @@ namespace swift {
 struct SymbolInfo {
 private:
 #if defined(_WIN32) && !defined(__CYGWIN__)
-  SYMBOL_INFO_PACKAGE _package;
+  const void *_symbolAddress;
+  const char *_symbolName;
+  const char *_moduleFileName;
+  const void *_moduleBaseAddress;
 #elif SWIFT_STDLIB_HAS_DLADDR
   Dl_info _info;
 #endif
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+  SymbolInfo(const void *symbolAddress,
+             const char *symbolName,
+             const char *moduleFileName,
+             const void *moduleBaseAddress)
+    : _symbolAddress(symbolAddress),
+      _symbolName(symbolName),
+      _moduleFileName(moduleFileName),
+      _moduleBaseAddress(moduleBaseAddress)
+  {}
+#endif
+
 public:
   SymbolInfo() {}
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+  SymbolInfo(const SymbolInfo &other) {
+    _symbolName = nullptr;
+    _moduleFileName = nullptr;
+    *this = other;
+  }
+  SymbolInfo(SymbolInfo &&other) {
+    *this = std::move(other);
+  }
+  ~SymbolInfo() {
+    if (_moduleFileName)
+      ::free((void *)_moduleFileName);
+    if (_symbolName)
+      ::free((void *)_symbolName);
+  }
+
+  SymbolInfo &operator=(const SymbolInfo &other) {
+    if (this != &other) {
+      if (_moduleFileName)
+        ::free((void *)_moduleFileName);
+      if (_symbolName)
+        ::free((void *)_symbolName);
```

Comment:
Likewise

Reply:
The reason I added it here is that I chose to implement the constructor using `*this = other;`, and I didn't like that calling `::free(nullptr)` twice every time on construction.

Reply:
I'll change it to not use `*this = other;` and to use a private member function instead, I think. That avoids my motivation for the tests.